src/pymovements/datasets/dataset.py

```diff
@@ -632,7 +679,10 @@ def save_preprocessed(self, preprocessed_dirname: str | None = None, verbose: in
                 print('Save file to', preprocessed_filepath)
 
             preprocessed_filepath.parent.mkdir(parents=True, exist_ok=True)
-            gaze_df_out.write_ipc(preprocessed_filepath)
+            if extension == 'feather':
+                gaze_df_out.write_ipc(preprocessed_filepath)
+            elif extension == 'csv':
+                gaze_df_out.write_csv(preprocessed_filepath)
```

Comment:
add an else clause and raise a `ValueError` in the `else` clause. use a text like this:
```
f'unsupported file format "{extension}". Supported formats are: {valid_extensions}'
```
with
```
valid_extensions = ['csv', 'feather']
```

Reply:
done